.github/actions/pre-commit-default/action.yml

```diff
@@ -0,0 +1,50 @@
+name: "Pre-commit format"
+description: "Install and run pre-commit"
+inputs:
+  pre-commit-args:
+    description: Additionnal parameters to pass to pre-commit
+    required: false
+  pre-commit-all-files:
+    description: >-
+      Whether to run pre-commit checks on all files (defaults to "true")
+    required: false
+    default: "true"
+  python-version:
+    description: The python version of the local runner
+    required: false
+    default: "3.9"
+  check-format:
+    description: >-
+      Whether to run format pre-commit checks.
+    required: false
+    default: "true"
+  check-github-configuration:
+    description: >-
+      Whether to run github actions, workflows, dependabot pre-commit checks.
+    required: false
+    default: "true"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
```

Comment:
not sure if we really want to checkout every time, this will impact workflows that are doing a previous actions/checkout with customized parameters (calling multiple times actions/checkout will just reset the current working tree)

Reply:
Would adding an input to avoid checkout (`do-checkout` true by default (?)) be enough?

Reply:
That's a good point @gionn , and IMO, this is one of the hardest part of creating GH actions: find the right granularity to avoid code duplication without being too opinionated or needing to declare all the possible customisations that called actions can provide.

In this specific case, can we assume that pre-commits will always going to have their dedicated job? If that's the case, I don't think we'll need customisations on checkout, as the customisations will be probably be done in another job.

Reply:
It's true that what I'm about to do is in caller workflows (whether they're in a dedicated job or not) is:
- call pre-commit action (leveraging the local pre-commit-config file if any)
- call pre-commit-default action (so here the checkout would be unneeded)

I did condition this checkout to a `do-checkout` parameter (true by default)

Reply:
> this is one of the hardest part of creating GH actions

it is indeed and I started to prefer to be less DRY and avoiding doing too much things in a single custom action. In the end is not really a problem to add an additional `- uses: actions/checkout@v3` in every workflow before using your custom action.

I am not against `do-checkout` input but I am not sure that including actions/checkout in a custom action is really worth the value (I mean, you are just saving a line in yaml).

Reply:
I agree and after thinking about it, it would make sense to remove it completely. Now we'll have an inconsistency between `pre-commit` and `pre-commit-default` actions: should this be a concern?

Reply:
After discussion, I removed the checkout and adjusted the README to make this explicit.